code-server GA doc updates (#562)
opendatahub-io/opendatahub-documentation: modules/installing-python-packages-on-your-notebook-server-in-code-server.adoc

:_module-type: PROCEDURE

[id="installing-python-packages-on-your-notebook-server-in-code-server_{context}"]
= Installing Python packages on your notebook server in code-server

[role='_abstract']

> You can install Python packages that are not part of the default notebook server by adding the package and the version to a `requirements.txt` file and then running the `pip install` command in a terminal window.

Reviewer: ```suggestion
You can install Python packages that are not part of the default workbench image by adding the package and the version to a `requirements.txt` file and then running the `pip install` command in a terminal window.
```

ifndef::upstream[]
NOTE: You can also install packages directly, but {org-name} recommends using a `requirements.txt` file so that the packages stated in the file can be easily re-used across different notebooks. In addition, using a `requirements.txt` file is also useful when using a S2I build to deploy a model.
endif::[]
ifdef::upstream[]
NOTE: You can also install packages directly, but using a `requirements.txt` file so that the packages stated in the file can be easily re-used across different notebooks is recommended. In addition, using a `requirements.txt` file is also useful when using a S2I build to deploy a model.
endif::[]

.Prerequisites
* A launched and running code-server notebook server.

.Procedure

> . In your code-server workbench, from the Activity Bar, select the menu icon (image:images/codeserver-menu-icon.png[The Menu icon]) > *File* > *New Text File*  to create a new text file.

Reviewer: ```suggestion
. In your code-server workbench, from the Activity Bar, select the menu icon (image:images/codeserver-menu-icon.png[The Menu icon]) > *File* > *New Text File* to create a new text file.
```

does the double-space make a difference, actually? maybe the rendered docs is the same with single or double space?

. Add the packages to install to the text file.
+
[source]
----
altair
----
+
You can specify the exact version to install by using the `==` (equal to) operator, for example:
+
[source]
----
altair==4.1.0
----
+
ifndef::upstream[]
[NOTE]
====
{org-name} recommends specifying exact package versions to enhance the stability of your notebook server over time. New package versions can introduce undesirable or unexpected changes in your environment's behavior.
====
endif::[]
ifdef::upstream[]
Specifying exact package versions to enhance the stability of your notebook server over time is recommended. New package versions can introduce undesirable or unexpected changes in your environment's behavior. 
endif::[]
To install multiple packages at the same time, place each package on a separate line.
. Save the text file as `requirements.txt`.
. From the Activity Bar, select the menu icon (image:images/codeserver-menu-icon.png[The Menu icon]) > *Terminal* > *New Terminal*  to open a terminal window.
. Install the packages in `requirements.txt` to your server by using the following command:
+
[source]
----
pip install -r requirements.txt
----
+
[IMPORTANT]
====

> This command installs the package on your notebook server, but you must still run the `import` directive to use the package in your code.

Reviewer: ```suggestion
This command installs the package on your workbench, but you must still run the `import` statement to use the package in your code.
```

Python calls these statements, c.f. https://docs.python.org/3/reference/import.html

----
import altair
----
====

.Verification
* Confirm that the packages in `requirements.txt` appear in the list of packages installed on the notebook server. 
ifndef::upstream[]

> See link:{rhoaidocshome}{default-format-url}/working_in_your_data_science_ide/#viewing-python-packages-installed-on-your-notebook-server-in-code-server_{context}[Viewing Python packages installed on your notebook server in code-server] for details.

Reviewer: ```suggestion
See link:{rhoaidocshome}{default-format-url}/working_in_your_data_science_ide/#viewing-python-packages-installed-on-your-notebook-server-in-code-server_{context}[Viewing Python packages installed on your code-server workbench] for details.
```

endif::[]
ifdef::upstream[]

> See link:{odhdocshome}/working-in-your-data-science-ide/#viewing-python-packages-installed-on-your-notebook-server-in-code-server_{context}[Viewing Python packages installed on your notebook server in code-server] for details.

Reviewer: ```suggestion
See link:{odhdocshome}/working-in-your-data-science-ide/#viewing-python-packages-installed-on-your-notebook-server-in-code-server_{context}[Viewing Python packages installed on your code-server workbench] for details.
```
